### DISCUSSION TITLE
opencv 코드질문

### DISCUSSION THREAD

**Discussion by SHEYCHAE** [kikiru328/Bone_Detection#32] [CHAT]
code에 올려둔 OpenCV_2에서 contour부분 자꾸 오류나는데 아시는분..~

**Comment by yibnn** [comment]
어디에서 오류나셧어요?? 잘은 모르지만 봐보겟습니다!!

**Reply by SHEYCHAE** [reply]
OpenCV_2 에 가면 drawContour부분이요..! 그부분 다 오류나요 ㅠ

**Reply by yibnn** [reply]
넵 지금 해볼게요!!

**Reply by yibnn** [reply]
오류 어떻게 나세요??
저 돌아가는것같아요
출력됬습니다

**Reply by SHEYCHAE** [reply]
코드 똑같이 돌리는데 되는건가요?? 
왜 제건안돼요.................................................................
error: OpenCV(4.1.2) /io/opencv/modules/imgproc/src/drawing.cpp:2509: error: (-215:Assertion failed) npoints > 0 in function 'drawContours'

**Reply by yibnn** [reply]
저도 저거 오류나서 찾아봤었는데 이미지가 제대로 안들어갔을때 저 오류가 난데요!! 경로 한번 확인해보세요!
저도 경로 고치니까 돌아갔습니다!! 현재 있는 폴더에 . 지정하고 돌려보세요!!
저는 이렇게 했습니다
cv2.imread("./dataset/contours.png", cv2.IMREAD_COLOR)

**Comment by SHEYCHAE** [comment]
if img.mean() <= 15:
        low = img.mean() * 1.5
        high = img.mean() * 1.6
    elif img.mean() <= 20:
        low = img.mean() * 1.5
        high = img.mean() * 1.8
    else:
        low = img.mean() * 1.5
        high = img.mean() * 2

    img = contrast_roi(img, low, high)

이 부분 code breaking하신 분 계신가요?? 
(contrast_roi 함수는 아래 첨부할게요)

**Reply by SHEYCHAE** [reply]
def contrast_roi(img, low, high):
    h, w = img.shape
    img_ = np.zeros(img.shape, dtype=np.uint8)

    for y in range(h):
        for x in range(w):
            temp = int((255 / (high - low)) * (img[y][x] - low))
            if temp > 255:
                img_[y][x] = 255
            elif temp < 0:
                img_[y][x] = 0
            else:
                img_[y][x] = temp

    return img_

**Reply by SHEYCHAE** [reply]
contrast_roi에서 temp 구하는 공식 왜저리 되는지도 잘 모르겠어요

**Reply by hye-jj** [reply]
컨투어 찾기 전에 contrast 사용되는 부분 저도 궁금했습니다!
사용된 식이 아래 블로글에서 설명되어 있는것 같습니다.  혹시 다른 자료를 찾으시면 공유 부탁드려요ㅎㅎ

https://marisara.tistory.com/entry/%ED%8C%8C%EC%9D%B4%EC%8D%AC-openCV-5-%EB%AA%85%EC%95%94%EB%8C%80%EB%B9%84-%EC%8A%A4%ED%8A%B8%EB%A0%88%EC%B9%ADstreching
